packages/stryker-babel-transpiler/src/BabelTranspiler.ts

```diff
@@ -1,69 +1,85 @@
+    this.projectRoot = this.determineProjectRoot(options);
     if (options.produceSourceMaps) {
       throw new Error(`Invalid \`coverageAnalysis\` "${options.config.coverageAnalysis}" is not supported by the stryker-babel-transpiler. Not able to produce source maps yet. Please set it to "off".`);
     }
-    this.knownExtensions = ['.js', '.jsx'];
   }
 
   public transpile(files: File[]): Promise<TranspileResult> {
-    const outputFiles: File[] = [];
     const errors: string[] = [];
-
-    files.forEach((file) => {
-      if (this.shouldBeTranspiled(file)) {
-        try {
-          outputFiles.push({
-            name: file.name,
-            content: this.transform((file as TextFile).content),
-            mutated: file.mutated,
-            included: file.included,
-            transpiled: file.transpiled,
-            kind: FileKind.Text
-          });
-        } catch (err) {
-          errors.push(`${file.name}: ${err.message}`);
-        }
-      } else {
-        outputFiles.push(file);
+    const transpiledFiles = files.map(file => {
+      try {
+        return this.transpileFileIfNeeded(file);
+      } catch (error) {
+        errors.push(`Error while transpiling "${file.name}": ${error.stack || error.message}`);
+        return file;
       }
     });
-
-    return Promise.resolve(this.createResult(outputFiles, errors));
+    return Promise.resolve(this.createResult(transpiledFiles, errors));
   }
 
-  private shouldBeTranspiled(file: File) {
-    return file.kind === FileKind.Text
-      && file.transpiled
-      && this.knownExtensions.indexOf(path.extname(file.name)) >= 0;
+  private transpileFileIfNeeded(file: File): File {
+    if (KNOWN_EXTENSIONS.some(ext => ext === path.extname(file.name))) {
+      return this.transpileFile(file);
+    } else {
+      return file; // pass through
+    }
   }
 
-  private transform(code: string) {
-    const result = babel.transform(code, this.babelConfig).code;
-
-    if (!result) {
-      throw new Error('Could not transpile file with the Babel transform function');
+  private transpileFile(file: File) {
+    const options = Object.assign({}, this.babelOptions, {
+      filename: file.name,
+      filenameRelative: path.relative(this.projectRoot, file.name)
+    });
+    const result = babel.transform(file.textContent, options);
+    if ((result as any).ignored) {
+      // Ignored will be true if the file was not transpiled (because it was ignored)
+      // TODO: Babel 7 will change this (according to a conversation I had on Slack).
+      //  => it will return a `null` value in that case
+      return file;
+    } else if (typeof result.code === 'undefined') {
+      throw new Error(`Could not transpile file "${file.name}". Babel transform function delivered \`undefined\`.`);
+    } else {
+      return new File(toJSFileName(file.name), result.code);
     }
+  }
 
-    return result;
+  private determineProjectRoot(options: TranspilerOptions): string {
```

Comment:
Should replace this with `config.basePath` when it's available https://github.com/stryker-mutator/stryker/issues/650

Reply:
True, that's why the babel transpiler is in the list.